File: mobile-client/android/UseCase/System/Analytic/RecordAnalyticEventUseCase.kt
package app.mobiling.client.client.usecase.system.analytic

import app.mobiling.client.client.contract.system.analytic.AnalyticEventPayload
import app.mobiling.client.client.data.system.analytic.AnalyticEventRecorder

/**
 * Canonical use case for analytics event emission.
 */
class RecordAnalyticEventUseCase(
    private val recorder: AnalyticEventRecorder = AnalyticEventRecorder(),
) {
    operator fun invoke(payload: AnalyticEventPayload): AnalyticEventPayload = recorder.record(payload)
}

File: client/ios/UseCase/System/Analytic/RecordAnalyticEventUseCase.swift
import Foundation

public struct RecordAnalyticEventUseCase {
    private let recorder: AnalyticEventRecorder

    public init(recorder: AnalyticEventRecorder = AnalyticEventRecorder()) {
        self.recorder = recorder
    }

    public func callAsFunction(_ payload: AnalyticEventPayload) -> AnalyticEventPayload {
        recorder.record(payload)
    }
}
